llms/mlx_lm/tuner/dora.py

```diff
@@ -75,16 +83,57 @@ def __init__(
             shape=(input_dims, r),
         )
         self.lora_b = mx.zeros(shape=(r, output_dims))
-        self.m = mx.linalg.norm(self.linear.weight, axis=1)
+
+    def set_linear(self, linear):
+        """
+        Set the self.linear layer and recompute self.m with respect to quantization
+        """
+        self.linear = linear
+        # self.m needs to be recomputed when self.linear changes
+        # which doesn't happen in from_linear when dora_lin.linear = linear
+        self.m = mx.linalg.norm(self._dequantized_weight(), axis=1).astype(mx.float32)
+
+    def _dequantized_weight(self):
+        """
+        Return the weight of linear layer and dequantize it if is quantized
+        """
+        weight = self.linear.weight
+        if self._is_quantized():
+            weight = mx.dequantize(
+                weight,
+                self.linear.scales,
+                self.linear.biases,
+                self.linear.group_size,
+                self.linear.bits,
+            )
+        return weight
+
+    def _is_quantized(self):
+        return isinstance(self.linear, nn.QuantizedLinear)
 
     def __call__(self, x):
         # Regular LoRA (without a bias)
```

Comment:
This comment should say `DoRA`?

Reply:
I assumed it said regular LoRA because DoRA is a type of low rank adaption, and the first part of this is basically regular LoRA. Lines 116 to 131 is standard LoRA without a bias.